### PR TITLE
Enable model testing (test set)

### PR DESCRIPTION
**Pull request by quentinverlhac**
This PR enables model testing after the last epoch.
It also changes the whole project structure to allow training and testing separately.

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+        for iteration, (spectro, label) in enumerate(train_dataset):
+            predictions, labels = train_step(spectro, label)
+
+            # display metrics
+            if iteration % 10 == 0:
+                display_and_reset_metrics(train_loss, train_accuracy, predictions, labels, iteration=iteration)
+
+            # manage checkpoint
+            checkpoint.iteration.assign(iteration)
```
Comment (Antoine-marchais) [inline train.py]: Not related to the main refacto, but could we use epochs instead of iterations ? when batch size is modified, it also modifies the checkpoint frequency.

File: test.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    for iteration, (spectro, label) in enumerate(test_dataset):
+        predictions, loss = forward_pass(spectro, label)
+
+        # compute metrics
+        test_loss.update_state(loss)
+        test_accuracy.update_state(label, predictions)
+
+    display_and_reset_metrics(test_loss, test_accuracy, predictions, label, is_test=True)
+
+if __name__ == '__main__':
```
Comment (Antoine-marchais) [inline test.py]: I think I would rather specify the name of the model I want to test using CLI args.
